### DISCUSSION TITLE
Can't connect to Victron Smartshunt

### DISCUSSION THREAD

**Discussion by Visnu7777** [fl4p/batmon-ha#221] [Q&A]
Hi @all,

I'm using batmon for my JKBMS with success, it works perfect. Now I added a Smartshunt to my system but batmon can't connect. I enabled GATT in the App, powercycled the shunt, cleared the cache in /var/lib/bluetooth, tried "000000", the actual pin and the Victron encryption key (:D) for pin in options.json but all I get is:

`10:43:44 INFO [main] Fetching 1 BMS + 0 virtual + 0 others serially, period=1.00s, keep_alive=True
10:43:44 INFO [sampling] connecting bms SmartShuntBt(D2:D3:AF:4E:BF:2B,smartshunt)
10:43:46 WARNING [victron] error sending keep alive b' N'
Traceback (most recent call last):
  File "/home/visnu/batmon-ha/bmslib/models/victron.py", line 65, in _keep_alive_loop
    await self.client.write_gatt_char('6597ffff-4bda-4c1e-af4b-551c4cf74769', data, response=False)
  File "/home/visnu/batmon-ha/venv/lib/python3.10/site-packages/bleak/backends/bluezdbus/client.py", line 945, in write_gatt_char
    assert_reply(reply)
  File "/home/visnu/batmon-ha/venv/lib/python3.10/site-packages/bleak/backends/bluezdbus/utils.py", line 23, in assert_reply
    raise BleakDBusError(reply.error_name, reply.body)
bleak.exc.BleakDBusError: [org.bluez.Error.NotPermitted] Not paired
10:43:46 ERROR [sampling] SmartShuntBt(D2:D3:AF:4E:BF:2B,smartshunt) device not found, retry in 1 seconds`

I also tried to remove the working JKBMS config temporarily but it makes no difference.
Any clues? Is there someone using batmon with the Smartshunt who can help me with firmware version numbers etc? Maybe victron changed the data structures?

Michael

**Comment by JoMeV** [comment]
Same situation in my system. Batmon is working with two JBD BMS but integration of Victron smartshunt doesn't work. I also would be happy for some useful hints.

**Reply by artclosson** [reply]
I just ran into a similar issue with a brand new BMV-712 smart shunt. It looks like GATT has been replaced with BLE advertisements. I have this working on a home assistant with another integration, but I would much rather use Batmon.  Here is a link to Victron's update:
https://communityarchive.victronenergy.com/questions/187303/victron-bluetooth-advertising-protocol.html.  I tried using the encryption key as the pin but it still cannot connect:  ERROR [sampling] SmartShuntBt(CE:81:F7:A5:xx:xx,SmartBMV HQ2310PT6CU) device not found, retry in 129 seconds
